src/spetlr/etl/loaders/simple_loader.py

```diff
@@ -15,20 +15,29 @@ def append(self, df: DataFrame) -> None:
         pass
 
 
+class Upsertable(Protocol):
```

Comment:
Maybe these protocols should live outside the simple loader, maybe a "load_modes.py" file in the "loaders directory" - just a suggestion